docs/web_apps.rst

```diff
@@ -188,17 +187,100 @@ Sessions
+
+It has top-level directories for the two major areas described above: ``form_entry`` for in-form behavior and
+``formplayer`` for the ``FormplayerFrontend`` application. There are also a few top-level directories and files for
+miscellaneous behavior.
+
+form_entry
+==========
+
+The `form_entry directory <https://github.com/dimagi/commcare-hq/tree/master/corehq/apps/cloudcare/static/cloudcare/js/form_entry>`_ contains the logic for viewing, filling out, and submitting a form.
+
+This is written in knockout, and it's probably the oldest code in this area.
+
+Major files to be aware of:
+
+* `form_ui.js <https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js>`_ defines ``Question`` and ``Container``, the major abstractions used by form definitions. ``Container`` is the base abstraction for groups and for forms themselves.
+* `entries.js <https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js>`_ defines ``Entry`` and its many subclasses, the widgets for entering data. The class hierarchy of entries has a few levels. There's generally a class for each question type: ``SingleSelectEntry``, ``TimeEntry``, etc. Appearance attributes can also have their own classes, such as ``ComboboxEntry`` and ``GeoPointEntry``.
+* `web_form_session.js <https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/cloudcare/static/cloudcare/js/form_entry/web_form_session.js>`_ defines the interaction for filling out a form. Web apps sends a request to formplayer every time a question is answered, so the session manages a lot of asynchronous requests, using a task queue. The session also handles loading forms, loading incomplete forms, and within-form actions like changing the form's language.
+
+Form entry has a fair amount of test coverage. There are entry-specific tests and also tests for web_form_session.
+
+formplayer
+==========
+
+The `formplayer directory <https://github.com/dimagi/commcare-hq/tree/master/corehq/apps/cloudcare/static/cloudcare/js/formplayer>`_
+contains logic for selecting an app, navigating through modules, displaying case lists, and almost everything besides filling out a form.
+
+This is written using Backbone and Marionette. Backbone is an MVC framework for writing SPAs, and Marionette is a library to simplify writing Backbone views.
+
+``FormplayerFrontend`` is the  "application" in this SPA.
+
+Miscellany
+==========
+
+This is everything not in either the ``form_entry`` or ``formplayer`` directory.
+
+debugger
+--------
+
+This controls the debugger, the "Data Preview" bar that shows up at the bottom of app preview and web apps and lets the user evaluate XPath and look at the form data and the submission XML.
+
+preview_app
+-----------
+
+This contains logic specific to app preview.
+
+There isn't much here: some initialization code and a plugin that lets you scroll by grabbing and dragging the app preview screen.
```

Comment:
Is the actual app preview code here?

Reply:
Very little of it - most of it is shared with web apps and is in the other directories.